python/ecl/summary/ecl_sum.py

```diff
@@ -524,6 +525,60 @@ def pandas_frame(self, time_index = None, column_keys = None):
         frame = pandas.DataFrame(index = time_index, columns=list(keywords), data=data)
         return frame
 
+    @staticmethod
+    def _compile_headers_list(headers, dims):
+        var_list = []
+        for key in headers:
+            lst = re.split(':', key)
+            kw = lst[0]
+            wgname = None
+            num = 0
+            unit = "UNIT"
+            if len(lst) > 1:
+                if lst[1][0].isdigit():
+                    nums = re.split(',', lst[1])
+                    if len(nums) > 1:
+                        i = int(nums[0])-1
+                        j = int(nums[1])-1
+                        k = int(nums[2])-1
+                        num = i + j * dims[0] + k * dims[0]*dims[1] + 1
+                    else:
+                        num = int(nums[0])                
+                else:
+                     wgname = lst[1]
+            var_list.append( [kw, wgname, num, unit] )     
+        return var_list   
+
+    @classmethod
+    def from_pandas(cls, case, frame, dims = (1,1,1), headers = None):
```

Comment:
I suggest that we use `dims = None` as default instead and the following `if` switch:
```
if dims is None:
     nx = 1
     ny = 1
     nz = 1
else:
     nx,ny,nz = dims
```
that way the dims is not accessed before it is actually needed in the `_compile_headers_list()` function. Could add a 
```
if dims is None:
     raise ValueError("When using ijk keywords like:{} you must supply a valid value for the dims argument".format(key))
```